### DISCUSSION TITLE
Best practice for setting custom stop word list non-globally

### DISCUSSION THREAD

**Discussion by pdhall99** [explosion/spaCy#12940] [Help: Best practices]
https://github.com/explosion/spaCy/discussions/10942 gives a recommendation for setting a custom stop word list. We might want to have several pipelines in the same session, some of which have default stop words, some of which have custom. Is there a better way to do this than the following, resetting to the default after loading?

```python
import spacy
from spacy.lang.en import EnglishDefaults

EN_DEFAULT_STOP_WORDS = frozenset(EnglishDefaults.stop_words)

# Load pipeline with custom stop_words
custom_stop_words = set(["hello", "world"])
EnglishDefaults.stop_words = custom_stop_words
nlp_custom = spacy.blank("en")
EnglishDefaults.stop_words = set(EN_DEFAULT_STOP_WORDS)  # Reset to defaults

# Load pipeline with default stop_words
nlp_default = spacy.blank("en")
```

**Comment by shadeMe** [comment]
Ideally, you'd create a [custom language subclass ](https://spacy.io/usage/linguistic-features#language-subclass) that uses a `CustomEnglishDefaults(BaseDefaults)` class with your stop words. You can refer to the [`English` language definition](https://github.com/explosion/spaCy/blob/master/spacy/lang/en/__init__.py) to see how that can be done.

If that's not an option, you can simplify the above code by eliminating the `EN_DEFAULT_STOP_WORDS` global - The default stopwords can be reloaded from `spacy.lang.en.stop_words.STOP_WORDS`.
